controllers/volumereplicationgroup_controller.go

```diff
@@ -577,3 +572,127 @@ func (v *VolumeReplicationGroupReconciler) SetupWithManager(mgr ctrl.Manager) er
 		Owns(&ramendrv1alpha1.VolumeReplicationGroup{}).
 		Complete(v)
 }
+
+// Upload the PV of the given PVC to object store, if the
+// VRG has been configured with an s3Endpoint.
+// - Return an error if S3Endpoint is configured but the PV
+//   could not be uploaded for any reason.
+// - If S3Endpoint is not configured, the user does not want
+//   VRG to backup/replicate the PV to an object store and has
+// 	 other means to replicate PV metadata and is likely using
+//   the VRG as a single control point to manipulate replication
+// 	 of a group of VRs and thus a group of PVs.  However, log
+//   a warning once per VRG
+func (v *VolumeReplicationGroupReconciler) uploadPV(
+	ctx context.Context,
+	volRepGroup *ramendrv1alpha1.VolumeReplicationGroup,
+	pvc *corev1.PersistentVolumeClaim) error {
+	vrgName := volRepGroup.Name
+
+	s3Endpoint := volRepGroup.Spec.S3Endpoint
+	if err := validateS3Endpoint(s3Endpoint, vrgName); err != nil {
+		return err
+	} else if s3Endpoint == "" {
+		// VRG is configured to run in a backup-less mode.  validateS3Endpoint()
+		// logs a warning message once about this backup-less mode; hence, no
+		// need to log any error here.
+		return nil
+	}
+
+	s3Conn, err := connectToS3Endpoint(ctx, v,
+		s3Endpoint,
+		types.NamespacedName{ /* secretName */
+			Name:      volRepGroup.Spec.S3SecretName,
+			Namespace: volRepGroup.Namespace,
+		},
+		vrgName, /* debugTag */
+	)
+	if err != nil {
+		return err
+	}
+
+	pv := corev1.PersistentVolume{}
+	volumeName := pvc.Spec.VolumeName
+	pvObjectKey := client.ObjectKey{
+		Name: volumeName,
+	}
+
+	if err := v.Get(ctx, pvObjectKey, &pv); err != nil {
+		return errorswrapper.Wrapf(err, "failed to get upload PV %v of VRG %v", volumeName, vrgName)
+	}
+
+	// Create the bucket, without caching or assuming its existence; use the VRG name
+	// as the bucket name
+	bucket := vrgName
```

Comment:
I think the bucket name should be the combination of namespace/vrgName.  vrgName is the subscription name and that name is not unique across all applications namespaces.

Reply:
Thanks, will do.